### ISSUE TITLE
Xbox One controllers no longer work wirelessly on Windows

### ISSUE THREAD

**Issue by kiates** [libsdl-org/SDL#5731]
Xbox One controllers no longer work via wireless on Windows using either the built-in support on Surface Book 2s or using the Microsoft Wireless Adapter. The controller is still detected but none of the inputs make it to the application. This worked at some point but I don't know which SDL version introduced the regression.

If the controller is connected via USB it works fine.

**Comment by slouken** [comment]
Is this with SDL you built yourself, or the official 2.0.20 release binaries?

**Comment by kiates** [comment]
Hi slouken,

I get it from dependencies for FNA, but I just diffed it and it is identical to the official 2.0.22 release binaries.

**Comment by kiates** [comment]
I'm trying to build main from source to test the latest and greatest, plus build the testgamecontroller utility to maybe debug further, but getting:

`1>SDL_windows.c
1>c:\program files (x86)\windows kits\8.1\include\winrt\roapi.h(56): error C2016: C requires that a struct or union has at least one member
1>Done building project "SDL.vcxproj" -- FAILED.`

when trying to build the VisualC\SDL\SDL.vcxproj.  Is there a work-around for that compiler issue besides hacking the windows SDK?

**Comment by kiates** [comment]
Hacked the roapi.h header to add a dummy field to the struct and can build SDL2 and testgamecontroller.  The issue reproduces there as well. Detects the controller when I turn it on, but not input detected when pressing buttons/triggers/etc:
![image](https://user-images.githubusercontent.com/1494951/170842071-4e1cf5be-e551-408e-bc0d-117d43c91a00.png)

**Comment by kiates** [comment]
I'm getting inconsistent results on this.  I originally discovered this on my Surface Book 2 running Windows 11.  The Surface Book 2 has built-in support for Xbox One Wireless tech (before Bluetooth versions).  I repro'd on my Windows 10 desktop using the Xbox Wireless Adapter.  But now after a reboot on the Windows 10 machine, it appears to be working fine.

I am still able to repro it on the Surface Book 2 and it now isn't working in either Wireless/Wired mode, but I'm going to reboot that as well, as this leads to me to believe it may be an environmental glitch/problem.  I'm not familiar with the innards of the joystick interface in SDL, but when I debugged into it, it was skipping HIDAPI stuff and going down the raw input path.  At that time I could get some action from the triggers only, but it was intermittent and only if I started by pressing both triggers at the same time.  Leads me to believe there was some cross-talk/corruption in the data it was reading. I'll continue to fiddle with the that box.

Long story short is it may not be a problem after all.

**Comment by kiates** [comment]
Rebooted the Surface Book 2 (it had a pending restart queued), and things are working fine on it as well in both the SDL testgamecontroller app and my use case.  I have no idea how I was able to get two machine to exhibit the same behavior.  Maybe they both had queued updates that needed restarts and there was some hiccup being in that state.

**Comment by Squall-Leonhart** [comment]
This is caused by a flaw in Windows Gaming Input / Gaming Input when a second user session is created on the system,

This can either be a result of an remote RDP login, or switching users, the only known resolution till now is to log out and back in.
